test/modules/TestResultCoordinator/report/TwinCountingReportGenerator.cs

```diff
@@ -46,13 +47,14 @@ public async Task<ITestResultReport> CreateReportAsync()
             while (await this.expectedTestResults.MoveNextAsync())
             {
                 Option<TwinTestResult> twinTestResult = this.GetTwinTestResult(this.expectedTestResults.Current);
+                Logger.LogDebug($"Expected test results {twinTestResult}");
 
                 twinTestResult.ForEach(
                     r =>
                     {
                         foreach (var prop in r.Properties)
                         {
-                            propertiesUpdated.Add(prop.ToString(), this.expectedTestResults.Current.CreatedAt);
+                            propertiesUpdated.TryAdd(prop.ToString(), this.expectedTestResults.Current.CreatedAt);
```

Comment:
The reason to use TryAdd here because avoiding crash if duplicates really happen, do I understand correctly?

Reply:
yes, that's correct